services/contract_call.proto

```diff
@@ -41,9 +41,19 @@ import "basic_types.proto";
  */
 message ContractCallTransactionBody {
     /**
-     * the contract instance to call, in the format used in transactions
+     * The contract to call
      */
-    ContractID contractID = 1;
+    oneof contract {
```

Comment:
Afaik, it will not be possible for a contract created on shard `0` to interact with a contract created in shard `1`.

Let me double-check with Leemon.